mlflow/pytorch.py

```diff
@@ -140,11 +141,13 @@ def save_model(pytorch_model, path, conda_env=None, mlflow_model=Model(), **kwar
     model_file = os.path.basename(model_path)
 
     conda_env_subpath = "conda.yaml"
-    if conda_env is not None:
-        shutil.copyfile(conda_env, os.path.join(path, conda_env_subpath))
-    else:
-        with open(os.path.join(path, conda_env_subpath), "w") as f:
-            yaml.safe_dump(DEFAULT_CONDA_ENV, stream=f, default_flow_style=False)
+    if conda_env is None:
+        conda_env = DEFAULT_CONDA_ENV
+    elif not isinstance(conda_env, dict):
+        with open(os.path.join(path, conda_env), "r") as f:
+            conda_env = yaml.safe_load(f)
+    with open(os.path.join(path, conda_env_subpath), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
 
     mlflow_model.add_flavor(FLAVOR_NAME, model_data=model_file, pytorch_version=torch.__version__)
     pyfunc.add_to_model(mlflow_model, loader_module="mlflow.pytorch",
```

Comment:
BTW, it seems like we could also implement dict-handling logic in `pyfunc.add_to_model` (i.e. allow `env` here to be a dictionary or a path to a file), which would allow us to avoid duplicating the logic across different flavor implementations. Seems like we still need to update the docs across the flavors, but it'd make it easier to update the logic later on - let me know what you think.